File: IsSubsequence/IsSubsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        counter = 0
        for i in t:
            if counter == len(s):
                break
            elif s[counter] == i:
                counter += 1
        return counter == len(s)


class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        i, j  = 0, 0
        while i < len(s) and j < len(t):
            if s[i] == t[j]:
                i += 1
            j += 1
        return i == len(s)

File: IsSubsequence/IsSubsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function (s, t) {
  let counter = 0;
  for (let i of t) {
    if (i === s[counter]) {
      counter++;
    }
  }
  return counter === s.length;
};


/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function (s, t) {
  let i = 0,
    j = 0;
  while (j < t.length) {
    if (s[i] === t[j]) {
      i++;
    }
    j++;
  }
  return i === s.length;
};
